### ISSUE TITLE
False-negative with `TC003`

### ISSUE THREAD

**Issue by osauldmy** [astral-sh/ruff#20308]
### Summary

Example to reproduce. In this case `Any` could be moved into `if TYPE_CHECKING` block, as it's not runtime-required import.

```python
from __future__ import annotations

from typing import TYPE_CHECKING, Any

if TYPE_CHECKING:
    from collections.abc import Iterator

def some_function(value: Any) -> Iterator[int]:
    del value
    yield from range(5)
```

If I remove the `Iterator` import outside, similarly to `Any`, `TC003` is correctly fired.

```python
from __future__ import annotations

from collections.abc import Iterator
from typing import TYPE_CHECKING, Any
```

`TC003 Move standard library import `collections.abc.Iterator` into a type-checking block`

Should also happen to `Any` from first snippet.

### Version

0.12.12

**Comment by second-ed** [comment]
Is your expected behaviour this?

```python
from __future__ import annotations

from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from collections.abc import Iterator
    from typing import Any
```

**Comment by osauldmy** [comment]
Yes

**Comment by ntBre** [comment]
`typing` is one of the default [exempt-modules](https://docs.astral.sh/ruff/settings/#lint_flake8-type-checking_exempt-modules). If you set that to `[]`, you get the behavior you expect: [playground](https://play.ruff.rs/1cef8b0a-1259-49a1-b52c-68c36c2c666f).

**Comment by osauldmy** [comment]
One more setting I didn't know about. Thanks. Solves my problem, closing the issue.
